tests/test_losses.py

```diff
@@ -134,22 +205,53 @@ def test_negative_sampling_self_adversarial_loss(self):
         # expected_loss ≈ 0.34
         expected_loss = (-pos_loss - neg_loss) / 2.
 
-        loss = loss_fct(pos_scores, neg_scores).item()
+        loss = self.instance(pos_scores, neg_scores).item()
 
         self.assertAlmostEqual(expected_loss, 0.34, delta=0.02)
         self.assertAlmostEqual(expected_loss, loss, delta=0.02)
 
-    def test_pipeline(self):
-        """Test the pipeline on RotatE with negative sampling self adversarial loss and nations."""
-        loss = NSSALoss
-        loss_kwargs = {"margin": 1., "adversarial_temperature": 1.}
-        pipeline_results = pipeline(
-            model='RotatE',
-            dataset='nations',
-            loss=loss,
-            loss_kwargs=loss_kwargs,
+
+class PairwiseLossTestsTest(TestsTest[PairwiseLoss], unittest.TestCase):
+    """unittest for unittests for pairwise losses."""
+
+    base_cls = PairwiseLoss
+    base_test_cls = _PairwiseLossTests
+
+
+class _SetwiseLossTests(_LossTests):
+    """unittests for setwise losses."""
+
+    instance: SetwiseLoss
+
+    #: Setwise do not support owa training loop
+    training_loop_support = dict(
+        owa=False,
```

Comment:
should be using slcwa and lcwa terminology now?

Reply:
Fixed in da4c3e3